src/libs/conduit/conduit_data_array.cpp

```diff
@@ -136,6 +139,87 @@ DataArray<T>::element(index_t idx) const
     return (*(T*)(element_ptr(idx)));
 }
 
+//---------------------------------------------------------------------------//
+template <typename T> 
+bool
+DataArray<T>::compatible(const DataArray<T> &array) const 
+{ 
+    return dtype().compatible(array.dtype());
+}
+
+//---------------------------------------------------------------------------//
+template <typename T> 
+bool
+DataArray<T>::equals(const DataArray<T> &array, const Node &config) const 
+{ 
+    Node info;
+    return !diff(array, info, config);
+}
+
+//---------------------------------------------------------------------------//
+template <typename T> 
+bool
+DataArray<T>::diff(const DataArray<T> &array, Node &info, const Node &config) const 
+{ 
+    // TODO(JRC): Consider moving the default machine epsilon to a constant
+    // value somewhere more generally accessible within Conduit.
+    float64 epsilon = config.has_child("eps") ? config["eps"].as_float64() : 1e-12;
+
```

Comment:
I think this is handy, but maybe we can just pass as a float64 and avoid the node?

Reply:
That's a good thought. I'd considered doing that originally, but I went with a `Node` to prevent tying the user-specified epsilon to any given type and in case there were more options in the future. I think though that it's best not to so predictive in this case as it's at the expense of usability (and the epsilon gets cast ultimately to a `float64` anyway), so I'm applying your suggestion in my next update.